libp2p/protocols/secure/noise.nim

```diff
@@ -99,10 +99,10 @@ type
 
 func shortLog*(conn: NoiseConnection): auto =
   try:
-    if conn.isNil: "NoiseConnection(nil)"
+    if conn == nil: "NoiseConnection(nil)"
     else: &"{shortLog(conn.peerId)}:{conn.oid}"
   except ValueError as exc:
-    raise newException(Defect, exc.msg)
+    raiseAssert(exc.msg)
```

Comment:
should we raise a `Defect` here?

Reply:
`raiseAssert` is very similar to `raise newException(Defect)`